format/FlightSql.proto

```diff
@@ -1779,9 +1796,60 @@ message CommandPreparedStatementUpdate {
 }
 
 /*
- * Returned from the RPC call DoPut when a CommandStatementUpdate
- * CommandPreparedStatementUpdate was in the request, containing
- * results from the update.
+ * Represents a bulk ingestion request. Used in the command member of FlightDescriptor
+ * for the the RPC call DoPut to cause the server load the contents of the stream's
+ * FlightData into the target destination.
+ */
+message CommandStatementIngest {
+  option (experimental) = true;
+
+  // Describes the behavior for loading bulk data.
+  enum IngestMode {
+    // Ingestion behavior unspecified.
+    INGEST_MODE_UNSPECIFIED = 0;
+    // Create the target table. Fail if the target table already exists.
+    INGEST_MODE_CREATE = 1;
+    // Append to an existing target table. Fail if the target table does not exist.
+    INGEST_MODE_APPEND = 2;
+    // Drop the target table if it exists. Then follow INGEST_MODE_CREATE behavior.
+    INGEST_MODE_REPLACE = 3;
+    // Create the target table if it does not exist. Then follow INGEST_MODE_APPEND behavior.
+    INGEST_MODE_CREATE_APPEND = 4;
+  }
+
+  // The ingestion behavior.
+  IngestMode mode = 1;
+  // The table to load data into.
+  string table = 2;
+  // The db_schema of the destination table to load data into. If unset, a backend-specific default may be used.
+  optional string schema = 3;
+  // The catalog of the destination table to load data into. If unset, a backend-specific default may be used.
+  optional string catalog = 4;
+  /*
+   * Store ingested data in a temporary table.
+   * Leaving temporary unspecified is equivalent to temporary=false.
+   * The effect of setting temporary depends on the choice of IngestMode:
```

Comment:
did you consider modelling this as two separate enums:

- Behavior on non-existing table.
- behavior on existing table.

?

Reply:
Antoine asked for this to be spelled out, but really this is just a longwided way to say that temporary=true is effectively a different way to specify a particular, vendor-specific catalog/schema corresponding to 'temporary tables' for that vendor, and otherwise the semantics are the same as for any other case where you have two tables with the same name in different namespaces (i.e. the tables are entirely unrelated)

Reply:
I think this is orthogonal.  I was asking about a modelling like:

```
enum TableDoesNotExistOptions {
   CREATE,
   FAIL
}
enum TableExistsOptions {
   FAIL,
   APPEND,
   REPLACE
}
```

Reply:
That sounds good to me.